lib/processors/minifier.js

```diff
@@ -28,11 +28,29 @@ function getPool(taskUtil) {
 			workerType: "auto",
 			maxWorkers
 		});
-		taskUtil.registerCleanupTask(() => {
-			log.verbose(`Terminating workerpool`);
-			const poolToBeTerminated = pool;
-			pool = null;
-			poolToBeTerminated.terminate();
+		taskUtil.registerCleanupTask((force) => {
+			const attemptPoolTermination = () => {
+				log.verbose(`Attempt to terminate the workerpool...`);
+
+				if (!pool) {
+					return;
+				}
+
+				// There are many stats that could be used, but these ones seem the most
+				// convenient. When all the (available) workers are idle, then it's safe to terminate.
+				const {idleWorkers, totalWorkers} = pool.stats();
+				if (idleWorkers !== totalWorkers && !force) {
+					return new Promise((resolve) =>
+						setTimeout(() => resolve(attemptPoolTermination()), 100) // Retry after a while
```

Comment:
There's no error propagation when `attemptPoolTermination()` fails synchronously, as the `reject` function of the `new Promise` is not called.
Can't you just use the promise variant of setTimeout here (`node:timers/promises`)?
The recursion might even be replaced with a loop (I might be wrong about that).

Reply:
Thanks!
Indeed, I was able to remove the recursion that way. Seems more clear now.